### PR TITLE
Add JTango recipe

### PR DESCRIPTION
**Pull request by beenje**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by beenje** [comment]
Hello @conda-forge/help-java, this is the first time I package some java.
The goal is to package at least 2 java applications that depend on some jar. I wanted to start with this jar first.
I read https://github.com/conda-forge/conda-forge.github.io/issues/590 and my understanding is that it's OK to package directly a jar file.

I also didn't find any convention where to put the jar file. I chose usr/share/java because this is where the jar are in some RPM we build.

I have questions regarding the license. [JTango](https://github.com/tango-controls/JTango) itself is released under LGPL-3.0 but the jar contains some other dependencies. A quick grep showed me some Apache and GPL-2 licenses. How does this usually work for jar files?
Under which license should this package be?

**Comment by beenje** [comment]
@gwen-soleil you maybe have some input about the JTango.jar license?

**Comment by wolfv** [comment]
awesome stuff. 

Is /usr/share/java on some sort of `JAVA...` path so that JTango could be found automatically? I don't know too much about Java and how it works in this case. 

We do have a bunch of java packages. I made one `openrefine` (although it's not noarch generic).

I think if you want to adhere to the rules, all libraries that are vendored in need their license packaged. Maybe JTango already does that? (they should!).

File: recipes/jtango/build.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+mkdir -p ${PREFIX}/usr/share/java
```
Comment (xhochy) [inline recipes/jtango/build.sh:1]: Omit the `usr` part.
```suggestion
mkdir -p ${PREFIX}/share/java
```

File: recipes/jtango/build.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+
+install -m 0644 JTango-${PKG_VERSION}.jar ${PREFIX}/usr/share/java
+ln -s JTango-${PKG_VERSION}.jar ${PREFIX}/usr/share/java/JTango.jar
```
Comment (xhochy) [inline recipes/jtango/build.sh:4]: I was able to generate a list of third party notices in the git tree using the following:

```suggestion
ln -s JTango-${PKG_VERSION}.jar ${PREFIX}/usr/share/java/JTango.jar

mvn install
mvn license:aggregate-third-party-report
```
Then include `target/site/aggregate-third-party-report.html` as part of the `license` field.

File: recipes/jtango/build.sh (hunk 1 above)
Comment (beenje) [reply recipes/jtango/build.sh]: Changed!

**Comment by beenje** [comment]
Thanks for the feedback!
I run `mvn license:aggregate-third-party-report` to get all the licenses and extracted the list in the `license` field.
I also added the text of the licenses in the `license_file` + the `aggregate-third-party-report.html` file.
I'm not sure that's how you suggested to do it.

File: recipes/jtango/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    - openjdk >=8
+
+test:
+  commands:
+    - test -f ${PREFIX}/share/java/JTango.jar
+    - test -f ${PREFIX}/share/java/JTango-${PKG_VERSION}.jar
+
+about:
+  home: https://www.tango-controls.org
+  license: Apache-2.0, MIT, SMLNJ, EPL-1.0, BSD-2-Clause, BSD-3-Clause, LGPL-2.1-or-later, LGPL-3.0-or-later, MPL-2.0
```
Comment (xhochy) [inline recipes/jtango/meta.yaml:27]: It should be sufficient to put the final license here and not all sublicenses, we mainly need the notices of the dependencies listed in the `license_file` section.

**Comment by xhochy** [comment]
> Thanks for the feedback!
> I run `mvn license:aggregate-third-party-report` to get all the licenses and extracted the list in the `license` field.
> I also added the text of the licenses in the `license_file` + the `aggregate-third-party-report.html` file.
> I'm not sure that's how you suggested to do it.

Could you do this as part of the build? Then we wouldn't need to redo this manually on every release.

File: recipes/jtango/meta.yaml (hunk 3 above)
Comment (beenje) [reply recipes/jtango/meta.yaml:27]: OK, so to make sure I understand, the following should be enough?

```yaml
license: LGPL-3.0-or-later
license_file:
  - LICENSE
  - aggregate-third-party-report.html
```

Comment (xhochy) [reply recipes/jtango/meta.yaml:27]: Yes

**Comment by beenje** [comment]
I updated the recipe to build the jar from the archive source.
I also generate the third party license report in the build script.

**Comment by beenje** [comment]
@xhochy would you have time to check my change? Thanks!

**Pull request merged**